Source/WebCore/workers/service/context/ServiceWorkerInspectorProxy.cpp

```diff
@@ -59,15 +61,20 @@ void ServiceWorkerInspectorProxy::serviceWorkerTerminated()
     m_channel = nullptr;
 }
 
-void ServiceWorkerInspectorProxy::connectToWorker(FrontendChannel& channel)
+void ServiceWorkerInspectorProxy::connectToWorker(FrontendChannel& channel, ServiceWorkerDebuggable& debuggable, bool isAutomaticConnection, bool immediatelyPause)
 {
     m_channel = &channel;
 
     RefPtr serviceWorkerThreadProxy = m_serviceWorkerThreadProxy.get();
     SWContextManager::singleton().setAsInspected(serviceWorkerThreadProxy->identifier(), true);
-    serviceWorkerThreadProxy->thread().runLoop().postDebuggerTask([] (ScriptExecutionContext& context) {
-        downcast<WorkerGlobalScope>(context).inspectorController().connectFrontend();
-    });
+    serviceWorkerThreadProxy->thread().runLoop().postDebuggerTask(
+        [weakDebuggable = ThreadSafeWeakPtr { debuggable }, isAutomaticConnection, immediatelyPause](ScriptExecutionContext& context) {
+            Function<void()> handleFrontendInitialized = [weakDebuggable] {
+                if (RefPtr debuggable = weakDebuggable.get())
+                    debuggable->unpauseForInitializedInspector();
```

Comment:
i think you might be able to get to the `ServiceWorkerDebuggable` from within the `WorkerInspectorController` via something like `m_globalScope->workerOrWorkletThread()->workerDebuggerProxy()` and then create a new `virtual void unpauseForInitializedInspector();` that does this

i just find passing in a lambda to be really brittle and confusing and hard to follow :(

Reply:
I'll do that plumbing work in a follow-up task. There might be a cleaner way out by casting the global scope into a `ServiceWorkerGlobalScope` and getting its `thread`.